src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2846,6 +2846,14 @@ private bool TryGetReturnType(out TypeWithAnnotations type, out FlowAnalysisAnno
         public override BoundNode? VisitLocal(BoundLocal node)
         {
             var local = node.LocalSymbol;
+
+            // Avoid cycles in nullable analysis.
+            if (local is SourceLocalSymbol { IsVar: true } && local.ForbiddenZone?.Contains(node.Syntax) == true)
+            {
+                SetResult(node, TypeWithState.ForType(node.Type), LvalueResultType, isLvalue: true);
```

Comment:
I'm not entirely sure what to call here. I don't really care for the result of the nullable analysis since this is an erroneous case. I just want to end the visit early.

Reply:
The lvalue state seems incorrect. Perhaps call `SetResult(node, TypeWithState.ForType(node.Type));` instead.

Reply:
Alright, such overload does not exist; but what about this? 9e2cb23a6b49d6dce925fc3536057d3c9830942c

Reply:
Sorry, I meant `SetResultType` rather than `SetResult`: 
```
SetResultType(node, TypeWithState.ForType(node.Type));`
```

Reply:
Oh, right, that makes sense. Changed: 75de5a29d374f2bffc25a99a01784b58a866f6ab